src/main/java/org/folio/entlinks/service/consortium/propagation/ConsortiumAuthoritySourceFilePropagationService.java

```diff
@@ -12,21 +13,26 @@
 public class ConsortiumAuthoritySourceFilePropagationService extends ConsortiumPropagationService<AuthoritySourceFile> {
 
   private final AuthoritySourceFileService sourceFileService;
+  private BiConsumer<AuthoritySourceFile, AuthoritySourceFile> updatePublishConsumer;
 
   public ConsortiumAuthoritySourceFilePropagationService(AuthoritySourceFileService sourceFileService,
                                                          ConsortiumTenantsService tenantsService,
                                                          SystemUserScopedExecutionService executionService) {
     super(tenantsService, executionService);
     this.sourceFileService = sourceFileService;
+    this.updatePublishConsumer = null;
   }
 
   protected void doPropagation(AuthoritySourceFile sourceFile, PropagationType propagationType) {
     switch (propagationType) {
       case CREATE -> sourceFileService.create(sourceFile);
-      case UPDATE -> sourceFileService.update(sourceFile.getId(), sourceFile);
+      case UPDATE -> sourceFileService.update(sourceFile.getId(), sourceFile, updatePublishConsumer);
       case DELETE -> sourceFileService.deleteById(sourceFile.getId());
       default -> throw new IllegalStateException("Unexpected value: " + propagationType);
     }
   }
 
+  public void setCurrentUpdatePublishConsumer(BiConsumer<AuthoritySourceFile, AuthoritySourceFile> consumer) {
```

Comment:
Maybe would be better to pass into propagation by using some composite object, f.e. `record SourceFilePropagationData(AuthoritySourceFile sourceFile, BiConsumer<AuthoritySourceFile, AuthoritySourceFile> consumer)` and extend `ConsortiumPropagationService<SourceFilePropagationData>`.
I don't think modifying internal state is a good idea.

Reply:
@viacheslavkol Thanks Slava, I've implemented this approach and put it to my baggage of knowledge :-)